From dd37a553ac60f16bdfc6f7d82532f0a6089add14 Mon Sep 17 00:00:00 2001
From: Mathieu Duponchelle <mathieu@centricular.com>
Date: Fri, 13 Jul 2018 17:24:07 +0200
Subject: [PATCH 48/72] av*dec: Ignore decoding errors

We were ignoring these before the port to 4.0, interpreting them
as GST_FLOW_ERROR / GST_ELEMENT_ERROR causes check failures.

We should start using GST_*_DECODER_ERROR in latter commits,
for now simply restore the previous behaviour.
---
 ext/libav/gstavauddec.c | 5 ++---
 ext/libav/gstavviddec.c | 5 ++---
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/ext/libav/gstavauddec.c b/ext/libav/gstavauddec.c
index 7810dc6..f202454 100644
--- a/ext/libav/gstavauddec.c
+++ b/ext/libav/gstavauddec.c
@@ -580,9 +580,8 @@ gst_ffmpegauddec_audio_frame (GstFFMpegAudDec * ffmpegdec,
     GST_WARNING_OBJECT (ffmpegdec,
         "Tried to receive frame on a flushed context");
   } else if (res < 0) {
-    *ret = GST_FLOW_ERROR;
-    GST_ELEMENT_ERROR (ffmpegdec, STREAM, DECODE, ("Decoding problem"),
-        ("Legitimate decoding error"));
+    *ret = GST_FLOW_OK;
+    GST_WARNING_OBJECT (ffmpegdec, "Legitimate decoding error");
   }
 
 beach:
diff --git a/ext/libav/gstavviddec.c b/ext/libav/gstavviddec.c
index e3ae04f..acb1ce9 100644
--- a/ext/libav/gstavviddec.c
+++ b/ext/libav/gstavviddec.c
@@ -1490,9 +1490,8 @@ gst_ffmpegviddec_video_frame (GstFFMpegVidDec * ffmpegdec,
         "Tried to receive frame on a flushed context");
     goto beach;
   } else if (res < 0) {
-    *ret = GST_FLOW_ERROR;
-    GST_ELEMENT_ERROR (ffmpegdec, STREAM, DECODE, ("Decoding problem"),
-        ("Legitimate decoding error"));
+    *ret = GST_FLOW_OK;
+    GST_WARNING_OBJECT (ffmpegdec, "Legitimate decoding error");
     goto beach;
   }
 
-- 
2.17.0

